### PR TITLE
add compat data for BatteryManager

### PR DESCRIPTION
**Pull request by gsouquet**
This should solve #968 
Are you aware of any other pages that need to be migrated, I'm happy to continue helping 👍

### CODE REVIEW

File: api/BatteryManager.json [hunk 1]
```diff
@@ -0,0 +1,850 @@
+{
+  "api": {
+    "BatteryManager": {
+      "__compat": {
+        "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager",
+        "support": {
+          "chrome": {
+            "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: This should be `"38"`

File: api/BatteryManager.json [hunk 2]
```diff
@@ -0,0 +1,850 @@
...
+            "version_added": false
+          },
+          "opera_android": {
+            "version_added": "25"
+          },
+          "safari_ios": {
+            "version_added": false
+          },
+          "chrome_android": {
+            "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 3]
```diff
@@ -0,0 +1,850 @@
...
+          "standard_track": true,
+          "deprecated": true
+        }
+      },
+      "charging": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/charging",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 4]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 5]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "chargingTime": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/chargingTime",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 6]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 7]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "dischargingTime": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/dischargingTime",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 8]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 9]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "level": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/level",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 39

File: api/BatteryManager.json [hunk 10]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 11]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "onchargingchange": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/onchargingchange",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 12]
```diff
@@ -0,0 +1,850 @@
...
+            "edge_mobile": {
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 13]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "onchargingtimechange": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/onchargingtimechange",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 14]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 15]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "ondischargingtimechange": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/ondischargingtimechange",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 16]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 17]
```diff
@@ -0,0 +1,850 @@
...
+            "deprecated": true
+          }
+        }
+      },
+      "onlevelchange": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/BatteryManager/onlevelchange",
+          "support": {
+            "chrome": {
+              "version_added": "39"
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

File: api/BatteryManager.json [hunk 18]
```diff
@@ -0,0 +1,850 @@
...
+              "version_added": false
+            },
+            "opera_android": {
+              "version_added": "25"
+            },
+            "safari_ios": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "42",
```
Comment (jpmedley) [inline api/BatteryManager.json]: 38

**Comment by gsouquet** [comment]
Updated everything you commented on @jpmedley . Thanks!

File: api/BatteryManager.json [hunk 19]
```diff
@@ -0,0 +1,850 @@
...
+            }
+          ],
+          "edge_mobile": {
+            "version_added": false
+          },
+          "opera_android": {
+            "version_added": "25"
+          },
+          "safari_ios": {
+            "version_added": false
```
Comment (teoli2003) [inline api/BatteryManager.json]: This one is a very special case: it has been added in an unknown version, and removed later in an unknown version (wasn't able to find which ones...)
So:
{
    "version_added": true,
    "version_removed": true
}

File: api/BatteryManager.json [hunk 20]
```diff
@@ -0,0 +1,850 @@
...
+          "chrome": {
+            "version_added": "38"
+          },
+          "firefox": [
+            {
+              "version_added": "10",
+              "version_removed": "16",
+              "prefix": "moz"
+            },
+            {
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply api/BatteryManager.json]: nit: we don't write with Gecko 18.0 (Firefox 18.0 / Thunderbird 18.0 / SeaMonkey 2.15) any more, replace this by Firefox 18. (And Firefox 11, without the .0)

**Comment by gsouquet** [comment]
@teoli2003 Thanks for your feedback really appreciated. I guess I have a better understanding on how to structure all of this. 

I guess all the requested changes are now fixed 👍 

How do you find all of those compat data in the first place? Is it just a case of you going through all the vendors changelog?

Thanks!

**Comment by Elchi3** [comment]
> How do you find all of those compat data in the first place? Is it just a case of you going through all the vendors changelog?

Yeah, source code, bug trackers, and changelogs mostly :-)

**Pull request merged**